torch/CMakeLists.txt

```diff
@@ -476,15 +476,237 @@ install(FILES
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES
+      ${TORCH_SRC_DIR}/lib/libshm)
+  endif()
+
+  set(TORCH_PYTHON_LINK_LIBRARIES
+    torch
+    shm)
+
+  set(TORCH_PYTHON_COMPILE_DEFINITIONS)
+
+  set(TORCH_PYTHON_COMPILE_OPTIONS)
+
+  set(TORCH_PYTHON_LINK_FLAGS)
+
+  if (MSVC)
+    list(APPEND TORCH_PYTHON_LINK_FLAGS " /NODEFAULTLIB:LIBCMT.LIB")
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${PYTHON_LIBRARIES})
+  elseif (APPLE)
+    list(APPEND TORCH_PYTHON_LINK_FLAGS " -undefined dynamic_lookup")
+  else()
+    list(APPEND TORCH_PYTHON_COMPILE_OPTIONS
+      -fno-strict-aliasing
+      -Wno-write-strings
+      -Wno-strict-aliasing)
+  endif()
+
+  if (USE_CUDA)
+    list(APPEND TORCH_PYTHON_SRCS
+      ${TORCH_SRC_DIR}/csrc/cuda/Module.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Storage.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Stream.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/utils.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/python_comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/serialization.cpp
+      ${TORCH_SRC_DIR}/csrc/nn/THCUNN.cpp
+      )
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_CUDA)
+
+    if(MSVC)
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${NVTOOLEXT_HOME}/lib/x64/nvToolsExt64_1.lib)
+      list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES "${NVTOOLEXT_HOME}/include")
+      list(APPEND TORCH_PYTHON_LINK_FLAGS " /FORCE:UNRESOLVED")
+    elseif(APPLE)
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib/libnvToolsExt.dylib)
+    else()
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib64/libnvToolsExt.so)
+    endif()
+  endif()
+
+  if (USE_CUDNN)
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_CUDNN)
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES ${CUDNN_INCLUDE_DIR})
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDNN_LIBRARY})
+  endif()
+
+  if (USE_MIOPEN)
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_MIOPEN)
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES ${MIOPEN_INCLUDE_DIR})
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${MIOPEN_LIBRARY})
+  endif()
+
+  if (USE_NUMPY)
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_NUMPY)
+  endif()
+
+  if (USE_ROCM)
+    list(APPEND TORCH_PYTHON_SRCS
+      ${TORCH_SRC_DIR}/csrc/cuda/Module.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Storage.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Stream.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/utils.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/python_comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/serialization.cpp
+      ${TORCH_SRC_DIR}/csrc/nn/THCUNN.cpp
+      )
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES caffe2_hip_library)
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS
+      USE_ROCM
+      __HIP_PLATFORM_HCC__
+      )
+  endif()
+
+  if (USE_DISTRIBUTED)
+    list(APPEND TORCH_PYTHON_SRCS ${TORCH_SRC_DIR}/csrc/distributed/Module.cpp)
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES ${TORCH_SRC_DIR}/lib/THD)
```

Comment:
what about c10d?

Reply:
c10d is modern cmake (THD is not), so it's taken care of by target_link_libraries below